packages/flutter/lib/src/material/text_selection_theme.dart

```diff
@@ -138,11 +141,29 @@ class TextSelectionTheme extends InheritedTheme {
     Key? key,
     required this.data,
     required Widget child,
-  }) : assert(data != null), super(key: key, child: child);
+  }) : assert(data != null),
+       _child = child,
+       super(key: key, child: const _NullWidget());
 
   /// The properties for descendant [TextField] and [SelectableText] widgets.
   final TextSelectionThemeData data;
 
+  // Overriding the getter to insert `DefaultSelectionStyle` into the subtree
+  // without a breaking change. This should be avoid in general since this
```

Comment:
breaking => breaking API

In general, this approach should be avoided because it relies on an implementation detail of ProxyWidget. This workaround is necessary because TextSelectionTheme is const.